lstchain/tools/lstchain_create_dl3_file.py

```diff
@@ -0,0 +1,201 @@
+
+    input_dl2 = traits.Path(
+        help="Input data DL2 file", exists=True, directory_ok=False, file_ok=True
+    ).tag(config=True)
+
+    output_dl3_path = traits.Path(
+        help="DL3 output filedir", directory_ok=True, file_ok=False
+    ).tag(config=True)
+
+    add_irf = traits.Bool(
+        help="True for adding IRF fits file to the DL3 file",
+        default_value=True,
+    ).tag(config=True)
+
+    input_irf = traits.Path(
+        help="Compressed FITS file of IRFs",
+        exists=True,
+        directory_ok=False,
+        file_ok=True,
+    ).tag(config=True)
+
+    source_name = traits.Unicode(help="Name of Source").tag(config=True)
+
+    config_file = traits.Path(
+        help="Config file for selection cuts",
+        directory_ok=False,
+        file_ok=True,
+    ).tag(config=True)
+
+    overwrite = traits.Bool(
+        help="If True, overwrites existing output file without asking",
+        default_value=True,
+    ).tag(config=True)
+
+    provenance_log = traits.Path(
+        help="Path for the Provenance log",
+        directory_ok=False,
+    ).tag(config=True)
+
+    aliases = {
+        "input_dl2": "DataReductionFITSWriter.input_dl2",
+        "d": "DataReductionFITSWriter.input_dl2",
+        "output_dl3_path": "DataReductionFITSWriter.output_dl3_path",
+        "o": "DataReductionFITSWriter.output_dl3_path",
+        "add_irf": "DataReductionFITSWriter.add_irf",
+        "input_irf": "DataReductionFITSWriter.input_irf",
+        "irf": "DataReductionFITSWriter.input_irf",
+        "config_file": "DataReductionFITSWriter.config_file",
+        "conf": "DataReductionFITSWriter.config_file",
+        "source_name": "DataReductionFITSWriter.source_name",
+        "provenance_log": "DataReductionFITSWriter.provenance_log",
+        "prov": "DataReductionFITSWriter.provenance_log",
+    }
+
+    flags = {
+        "overwrite": (
+            {"DataReductionFITSWriter": {"overwrite": True}},
+            "overwrite output file",
+        )
+    }
+
+    def __init__(self, **kwargs):
+        super().__init__(**kwargs)
+        self.data = None
+        self.filename_dl3 = None
+        self.run_number = None
+        self.effective_time = None
+        self.elapsed_time = None
+        self.cuts = None
+        self.events = None
+        self.gti = None
+        self.pointing = None
+        self.aeff2d = None
+        self.edisp2d = None
+        self.bkg2d = None
+        self.psf = None
+        self.hdulist = None
+        self.output_file = None
+
+    def setup(self):
+        if self.config_file is None:
+            self.cuts = read_configuration_file(
+                os.path.join(
+                    os.path.dirname(__file__), "../data/data_selection_cuts.json"
+                )
+            )
+        else:
+            self.cuts = read_configuration_file(self.config_file)
+
+        self.filename_dl3 = dl2_to_dl3_filename(self.input_dl2)
+
+        if not self.provenance_log:
```

Comment:
`provenance_log` will never be None I think, since it has a dynamic default. But again, why explicitly override the defaults provided by Tool?

Reply:
I wanted to have the option to save it either in a separate folder or in the same location where the DL3 files are generated. This was not the case by default, so I added this option.